stl/src/syncstream.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// initialize syncstream mutex map
+
+#include <internal_shared.h>
+#include <map>
+#include <shared_mutex>
+
+#pragma warning(disable : 4074)
+#pragma init_seg(compiler)
+
+_STD_BEGIN
+// OBJECT DECLARATIONS
+struct _Mutex_count_pair {
+    shared_mutex _Mutex;
+    uint64_t _Ref_count = 0;
+};
+
+template <class _Ty>
+class _Crt_allocator {
+public:
+    using value_type                             = _Ty;
+    using size_type                              = size_t;
+    using difference_type                        = ptrdiff_t;
+    using propagate_on_container_move_assignment = true_type;
+    using is_always_equal                        = true_type;
+
+    constexpr _Crt_allocator() noexcept = default;
+
+    constexpr _Crt_allocator(const _Crt_allocator&) noexcept = default;
+    template <class _Other>
+    constexpr _Crt_allocator(const _Crt_allocator<_Other>&) noexcept {}
+
+    _NODISCARD __declspec(allocator) _Ty* allocate(_CRT_GUARDOVERFLOW const size_t _Count) {
+        return static_cast<_Ty*>(_calloc_crt(_Count, sizeof(_Ty)));

```

Comment:
This function should probably check if `_calloc_crt` returns `nullptr` and throw `bad_alloc` if it does.